bin/brew

```diff
@@ -9,20 +9,43 @@ BREW_FILE_DIRECTORY="$(quiet_cd "${0%/*}/" && pwd -P)"
 HOMEBREW_BREW_FILE="${BREW_FILE_DIRECTORY%/}/${0##*/}"
 
 HOMEBREW_PREFIX="${HOMEBREW_BREW_FILE%/*/*}"
-[[ -n "$HOMEBREW_PREFIX" && "$HOMEBREW_PREFIX" != "$HOMEBREW_BREW_FILE" ]] \
-  || HOMEBREW_PREFIX="/"
+if [[ -z "$HOMEBREW_PREFIX" || "$HOMEBREW_PREFIX" == "$HOMEBREW_BREW_FILE" ]]
+then
+  HOMEBREW_PREFIX="/"
+fi
 
 HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX"
 
 if [[ -L "$HOMEBREW_BREW_FILE" ]]
 then
+  HOMEBREW_BREW_FILE_LINK="1"
+fi
+
+if [[ -L "/usr/local/bin/brew" ]]
+then
+  HOMEBREW_USR_LOCAL_HOMEBREW_TEST="1"
+fi
+
+if [[ -n "$HOMEBREW_BREW_FILE_LINK" ||
+      -n "$HOMEBREW_USR_LOCAL_HOMEBREW_TEST" ]]
+then
   BREW_SYMLINK="$(readlink "$HOMEBREW_BREW_FILE")"
```

Comment:
> Why is it necessary to run the code in this if block when HOMEBREW_USR_LOCAL_HOMEBREW_TEST is set? It doesn't seem to do anything useful.

To get `BREW_FILE_DIRECTORY` that is required for both cases but used differently.




> The result returned by this particular line is an empty string if HOMEBREW_USR_LOCAL_HOMEBREW_TEST is set

I should have temporarily set `HOMEBREW_BREW_FILE` to `/usr/local/bin/brew` before this which would have provided the behaviour I'm looking for without duplicating these three lines. I'm open to a better approach, though.